activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -746,10 +747,20 @@ def pk_and_sequence_for(table)
         end
       end
 
-      # Returns just a table's primary key
-      def primary_key(table)
-        pk_and_sequence = pk_and_sequence_for(table)
-        pk_and_sequence && pk_and_sequence.first
+      def primary_keys(table_name) # :nodoc:
+        return [] unless table_name.present?
+
+        schema, name = table_name.to_s.split('.', 2)
+        schema, name = @config[:database], schema unless name # A table was provided without a schema
+
+        select_values(<<-SQL.strip_heredoc, 'SCHEMA')
+          SELECT column_name
+          FROM information_schema.key_column_usage
+          WHERE constraint_name = 'PRIMARY'
+            AND table_schema = #{quote(schema)}
+            AND table_name = #{quote(name)}
+          ORDER BY ordinal_position
+        SQL
```

Comment:
`information_schema` is already used to [foreign_keys](https://github.com/kamipo/rails/blob/correctly_dump_composite_primary_key/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L677-L687).
`pk_and_sequence_for` returns a table's single column primary key and belonging sequence.
`primary_keys` returns a table's all primary key columns by the order.